src/contract-ui/tabs/edit-extensions/page.tsx

```diff
@@ -0,0 +1,149 @@
+import {
+  Stack,
+  Alert,
+  AlertIcon,
+  Flex,
+  Spacer,
+  Spinner,
+} from "@chakra-ui/react";
+import { InstalledExtensionsTable } from "./components/InstalledExtensionsTable";
+import { InstallExtensionForm } from "./components/ExtensionForm";
+import { Heading, Text } from "../../../tw-components";
+import { useEVMContractInfo } from "../../../@3rdweb-sdk/react";
+import { defineChain, getContract } from "thirdweb";
+import { getInstalledExtensions, owner } from "thirdweb/extensions/modular";
+import { useActiveAccount, useReadContract } from "thirdweb/react";
+import { thirdwebClient } from "../../../lib/thirdweb-client";
+import { useMemo } from "react";
+
+interface ContractEditExtensionsPageProps {
+  contractAddress?: string;
+}
+
+export const ContractEditExtensionsPage: React.FC<
+  ContractEditExtensionsPageProps
+> = ({ contractAddress }) => {
+  const contractInfo = useEVMContractInfo();
+
+  const chainId = contractInfo?.chain?.chainId;
+
+  if (!contractAddress || !chainId) {
+    return (
+      <Flex flex={1} alignItems="center" justifyContent="center" h={[300, 500]}>
+        <Spinner size="xl" />
+      </Flex>
+    );
+  }
+
+  return <Content contractAddress={contractAddress} chainId={chainId} />;
+};
+
+function Content(props: { contractAddress: string; chainId: number }) {
+  const { contractAddress, chainId } = props;
+  const account = useActiveAccount();
+
+  const contract = useMemo(
+    () =>
+      getContract({
+        client: thirdwebClient,
+        address: contractAddress,
+        chain: defineChain(chainId),
+      }),
+    [contractAddress, chainId],
+  );
+
+  const installedExtensionsQuery = useReadContract(getInstalledExtensions, {
+    contract,
+  });
+
+  const ownerQuery = useReadContract(owner, {
+    contract,
+  });
+
+  function refetchExtensions() {
+    installedExtensionsQuery.refetch();
+  }
+
+  if (!ownerQuery.data) {
```

Comment:
prefer `isLoading` state here, right now if there just was no return data this would spin forever (even once "loading" finished)